backend/layers/api/portal_api.py

```diff
@@ -430,6 +456,9 @@ def publish_post(self, collection_id: str, body: object, token_info: dict):
         return make_response({"collection_id": version.collection_id.id, "visibility": "PUBLIC"}, 202)
 
     def upload_from_link(self, collection_id: str, token_info: dict, url: str, dataset_id: str = None):
+        """
```

Comment:
I believe this method is used only for dropbox links (at least per the error handling), while the `upload_link` method below is used for presigned s3 URLs. If this is the case, should be noted in the comment

That said, it probably could be consolidated. But we don't need to do that here